carRental/src/main/java/com/amigoscode/chohort2/carRental/image/MultiMediaS3Handler.java

```diff
@@ -0,0 +1,96 @@
+package com.amigoscode.chohort2.carRental.image;
+
+import com.amigoscode.chohort2.carRental.external.s3.S3Buckets;
+import com.amigoscode.chohort2.carRental.external.s3.S3Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * @param <ID> the type of domain's id
+ * @author DmitriKonnovNN
+ * @since 0.0.1
+ */
+public interface MultiMediaS3Handler<ID> {
```

Comment:
I think you dont need this interface. as the name suggests `MultiMediaS3Handler` it is s3 specific